### PR TITLE
add ability to update PR by adding a label

### PR DESCRIPTION
**Pull request by denyskon**
This feature allows a maintainer to update a PR's branch by adding the `giteabot/update-branch` label. It is being removed after the update.

### CODE REVIEW

File: src/prActions.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    // if there is a merge conflict, we'll add a comment to fix the conflicts
+    await addComment(
+      pr.number,
+      `@${pr.user.login} please fix the merge conflicts. :tea:`,
+    );
+  }
+};
+
+// make sure we don't trigger too often
+export const run = debounce(execute, 8000);
```
Comment (yardenshoham) [inline src/prActions.ts:50]: Is this code copy-pasted? Can you refactor to avoid duplication? I can do it in another PR if you prefer

Comment (denyskon) [reply src/prActions.ts]: I tried my best.....

File: src/prActions.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  switch (label) {
+    case "giteabot/update-branch": {
+      const err = await updateBranch(pr);
+      if (err?.message == "failed to sync PR") {
+        // notify maintainer about failed update
+        await addComment(
+          pr.number,
+          `Failed to update branch :tea:`,
+        );
+      }
```
Comment (yardenshoham) [inline src/prActions.ts:17]: Logging the message is enough, don't comment

Comment (denyskon) [reply src/prActions.ts]: Okay, removed

File: src/prActions.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+const execute = async (
+  label: string,
+  pr: { number: number; user: { login: string } },
+) => {
+  switch (label) {
+    case "giteabot/update-branch": {
+      await updateBranch(pr);
+      await removeLabel(pr.number, "giteabot/update-branch");
+    }
+  }
```
Comment (silverwind) [inline src/prActions.ts:13]:
```suggestion
  if (label === "giteabot/update-branch") {
    await updateBranch(pr);
    await removeLabel(pr.number, "giteabot/update-branch");
  }
```

More of a personal preference, but I almost never use `switch` in Javascript. The only place I use it is in performance-critical parts as last I checked it is a bit faster then `if` branches.

Comment (denyskon) [reply src/prActions.ts]: I don't really have a preference, fine with me

File: src/prActions.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import { addComment, needsUpdate, removeLabel, updatePr } from "./github.ts";
+import { debounce } from "https://deno.land/std@0.189.0/async/mod.ts";
+
+const execute = async (
+  label: string,
+  pr: { number: number; user: { login: string } },
+) => {
+  if (label === "giteabot/update-branch") {
+    await updateBranch(pr);
+    await removeLabel(pr.number, "giteabot/update-branch");
```
Comment (yardenshoham) [inline src/prActions.ts:10]: If it doesn't need updating, maybe we post a comment before removing?

Comment (yardenshoham) [reply src/prActions.ts:10]: Oops sorry I requested the opposite above

**Pull request merged**